tests/unit-test/test_serializer.c

```diff
@@ -29,27 +30,83 @@ void test_serialize_ta_generate_address(void) {
 }
 
 void test_deserialize_ta_send_transfer(void) {
-  const char* json =
+  const char* json_template =
       "{\"value\":100,"
       "\"message_format\":\"trytes\","
-      "\"message\":\"" TEST_TAG "\",\"tag\":\"" TEST_TAG
+      "\"message\":\"%s\",\"tag\":\"" TEST_TAG
       "\","
       "\"address\":\"" TRYTES_81_1 "\"}";
-
+  tryte_t test_transfer_message[TEST_TRANSFER_MESSAGE_LEN + 1] = {};
+  gen_rand_trytes(TEST_TRANSFER_MESSAGE_LEN, test_transfer_message);
+  const int len = strlen(json_template) + TEST_TRANSFER_MESSAGE_LEN;
+  char* json = (char*)malloc(sizeof(char) * len);
+  snprintf(json, len, json_template, test_transfer_message);
   ta_send_transfer_req_t* req = ta_send_transfer_req_new();
-  flex_trit_t tag_msg_trits[FLEX_TRIT_SIZE_81], hash_trits_1[FLEX_TRIT_SIZE_243];
+  flex_trit_t tag_trits[NUM_FLEX_TRITS_TAG] = {}, hash_trits[NUM_FLEX_TRITS_HASH] = {},
+              msg_trits[NUM_FLEX_TRITS_MESSAGE] = {};
 
   ta_send_transfer_req_deserialize(json, req);
+  TEST_ASSERT_EQUAL_INT(100, req->value);
+  flex_trits_from_trytes(tag_trits, NUM_TRITS_TAG, (const tryte_t*)TEST_TAG, NUM_TRYTES_TAG, NUM_TRYTES_TAG);
+  flex_trits_from_trytes(msg_trits, NUM_TRITS_MESSAGE, test_transfer_message, NUM_TRYTES_MESSAGE, NUM_TRYTES_MESSAGE);
+  TEST_ASSERT_EQUAL_MEMORY(tag_trits, req->tag->hash, NUM_FLEX_TRITS_TAG);
+  TEST_ASSERT_EQUAL_MEMORY(msg_trits, req->message, NUM_FLEX_TRITS_MESSAGE);
+
+  flex_trits_from_trytes(hash_trits, NUM_TRITS_HASH, (const tryte_t*)TRYTES_81_1, NUM_TRYTES_HASH, NUM_TRYTES_HASH);
+  TEST_ASSERT_EQUAL_MEMORY(hash_trits, req->address->hash, NUM_FLEX_TRITS_HASH);
+
+  ta_send_transfer_req_free(&req);
+  free(json);
+}
+
+void test_deserialize_ta_send_transfer_raw_message(void) {
```

Comment:
What is the difference between normal and raw message?

Reply:
raw trytes will be encoded into trytes, so the length will be twice larger than raw trytes

Reply:
Is there any reference for raw trytes?
I am confused about when will TA get raw trytes?

Reply:
Typo I should said `raw message` instead of `raw tryte`
Raw message stands for message which hasn't been encoded into trytes, so it would be twice shorter than encoded into trytes